hs/src/Reach/Connector/ALGO.hs

```diff
@@ -2991,7 +3005,11 @@ cmeth sigi = \case
       comment $ LT.pack $ " ui: " <> show sigi
       let f :: DLType -> Integer -> (DLType, App ())
           f t i = (t, code "txna" ["ApplicationArgs", texty i])
-      cconcatbs_ (const $ return ()) $ zipWith f tys [1 ..]
+      -- If there are more than 15 args, args 15+ are packed as a tuple in arg 15.
+      let effectiveTys = case 15 < (length tys) of
+            False -> tys
+            True -> (take 14 tys) <> [T_Tuple $ drop 14 tys]
+      cconcatbs_ (const $ return ()) $ zipWith f effectiveTys [1 ..]
```

Comment:
This is very suspicious that it works, because the code that we're jumping to really expects to have all e.g. 20 arguments. I think it is just a coincidence that it works because the format on the single tuple is the same as the 20 all concatenated together. Thus, this code is not type-correct but happens to have the same representation. At the very least, you should comment on that

Reply:
Comment then